docs/core/compatibility/serialization/8.0/publishtrimmed.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "Breaking change: PublishedTrimmed projects fail reflection-based serialization"
+description: Learn about the .NET 8 breaking change in System.Text.Json serialization where projects with PublishTrimmed enabled now fail reflection-based serialization by default.
+ms.date: 11/08/2023
+---
+# PublishedTrimmed projects fail reflection-based serialization
+
+Projects that enable the [PublishTrimmed](../../../deploying/trimming/trimming-options.md#enable-trimming) MSBuild property now automatically turn off the reflection-based defaults of [System.Text.Json](../../../../standard/serialization/system-text-json/overview.md). In other words, setting `PublishTrimmed` to `true` automatically sets the `JsonSerializerIsReflectionEnabledByDefault` MSBuild property to `false` unless otherwise specified in project configuration.
+
+## Previous behavior
+
+Prior to this change, projects that have the `PublishTrimmed` property enabled, that is, `<PublishTrimmed>true</PublishTrimmed>` published a trimmed application. However, the reflection-based default serialization behavior wasn't necessarily disabled. In other words, depending on what code got trimmed, the follow code might or might not succeed serialization, or might or might not output the correct serialization data.
+
+```csharp
+JSonSerializer.Serialize(new { Value = 42 });
+```
+
+## New behavior
+
+Starting in .NET 8, projects that have the `PublishTrimmed` property enabled fail serialization outright. The code `JSonSerializer.Serialize(new { Value = 42 });` throws the following exception:
+
+> **System.InvalidOperationException: Reflection-based serialization has been disabled for this application.**
+
+## Version introduced
+
+.NET 8 Preview 7
+
+## Type of breaking change
+
+This change is a [behavioral change](../../categories.md#behavioral-change).
+
+## Reason for change
+
+This change ensures that trimmed applications use appropriate defaults. It also guides users towards adopting best practices suitable for trimmed applications&mdash;that is, use the source generator, and avoid accidental dependency on the unsafe reflection-based components.
+
+## Recommended action
+
+You can get back the original behavior by explicitly enabling the `JsonSerializerIsReflectionEnabledByDefault` property in your project file:
+
+```xml
+<PropertyGroup>
+  <PublishTrimmed>true</PublishTrimmed>
+  <JsonSerializerIsReflectionEnabledByDefault>true</JsonSerializerIsReflectionEnabledByDefault>
+</PropertyGroup>
+```
+
+However, we strongly recommend that you take appropriate action to migrate your trimmed applications to use the source generator.
```

Comment:
Can this be put higher up in the "Recommended action" section?

So it reads like: "we disabled it for good reasons, but you can just re-enable it".
Thus I'd put the move to source generator as first point in the "Recommended action", then if that's not possible enable the reflection based serialization.